### ISSUE TITLE
Memory leak caused by HealthCheckHandler function

### ISSUE THREAD

**Issue by khangtna** [kimdre/doco-cd#190]
Hi @kimdre,

When I deployed and handled a high volume of commits to git within a time frame, I noticed that the agent memory usage increased abnormally. 

<img width="1328" alt="Screenshot 2024-11-05 at 21 38 53" src="https://github.com/user-attachments/assets/c97bf3d1-fd2a-4a19-97c2-f79350148b1f">

`
After checking the code, I found that the **HealthCheckHandler()** function includes a segment that checks the Docker socket connection via the **VerifySocketConnection()** function. In this function, the **httpClient** instance is created but not closed after use.

`httpClient := NewHttpClient()`

When I added code to ensure the **httpClient** is released after use.

`defer httpClient.CloseIdleConnections()`

```go
httpClient := NewHttpClient()

err = VerifySocketRead(httpClient, apiVersion)
if err != nil {
  return err
}

defer httpClient.CloseIdleConnections() //close http client
```

It worked. Memory usage is now stable, with no further unexpected increases.
Please check this again.

**Comment by kimdre** [comment]
Thank you for noticing this. :)

**Comment by kimdre** [comment]
The fix is released with [v0.12.2](https://github.com/kimdre/doco-cd/releases/tag/v0.12.2)

**Comment by khangtna** [comment]
You're welcome. Thanks for your app, it's a great gitops CD.
